services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/links/IslCostSpec.groovy

```diff
@@ -131,4 +131,45 @@ class IslCostSpec extends BaseSpecification {
             assert islUtils.getIslInfo(links, reverseIsl).get().state == IslChangeType.DISCOVERED
         }
     }
+
+    def "ISL(not BFD) is NOT FAILED earlier than discoveryTimeout is exceeded when connection is lost(not port down)"() {
+        given: "ISL going through a-switch and state of ISL"
+        def isl = topology.islsForActiveSwitches.find {
+            it.aswitch?.inPort && it.aswitch?.outPort && !it.bfd
+        } ?: assumeTrue("Wasn't able to find suitable ISL", false)
+
+        def beforeState = islUtils.getIslInfo(isl).get().state
+        assert beforeState == IslChangeType.DISCOVERED
+
+        double waitTime = discoveryTimeout - (discoveryTimeout * 0.2)
+        double interval = discoveryTimeout * 0.2
+        def ruleToRemove = [isl.aswitch]
+        def reverseIsl = islUtils.reverseIsl(isl)
+
+        when: "Remove a one-way flow on an a-switch for simulating lost connection(not port down)"
+        lockKeeper.removeFlows(ruleToRemove)
+
+        then: "Status of ISL is not changed to FAILED until discoveryTimeout is exceeded"
+        Wrappers.timedLoop(waitTime) {
+            def links = northbound.getAllLinks()
+            assert islUtils.getIslInfo(links, isl).get().state == IslChangeType.DISCOVERED
+            assert islUtils.getIslInfo(links, reverseIsl).get().state == IslChangeType.DISCOVERED
+            sleep((interval * 1000).toLong())
+        }
+
+        and: "Status of ISL is changed to FAILED when discoveryTimeout is exceeded"
+        Wrappers.wait(WAIT_OFFSET, interval) {
+            def links = northbound.getAllLinks()
+            assert islUtils.getIslInfo(links, isl).get().state == IslChangeType.FAILED
+            assert islUtils.getIslInfo(links, reverseIsl).get().state == IslChangeType.FAILED
```

Comment:
I want you to add the `isl.actualState` verifications for `isl` and `reverseIsl`. This field was added just recently and we need to verify it is holding valid information. This test is the best place to check it

Reply:
Done.